subcommands/keys/tuf_rotate_offline_key.go

```diff
@@ -0,0 +1,380 @@
+Instead, please, use the "fioctl keys tuf rotate-offline-key --role=root --creds=<tuf-creds.tar.gz>".
+`,
+		Hidden:      true,
+		Run:         doRotateOfflineKey,
+		Annotations: map[string]string{tufCmdAnnotation: tufCmdRotateRootLegacy},
+		Args:        cobra.ExactArgs(1),
+	}
+	legacyRotateRoot.Flags().BoolP("sync-prod", "", false, "Make sure production root.json is up-to-date and exit")
+	legacyRotateRoot.Flags().BoolP("initial", "", false, "Used for the first customer rotation. The command will download the initial root key")
+	legacyRotateRoot.Flags().StringP("changelog", "m", "", "Reason for doing rotation. Saved in root metadata for tracking change history")
+	legacyRotateRoot.Flags().StringP("key-type", "k", tufKeyTypeNameRSA, "Key type, supported: Ed25519, RSA (default).")
+	cmd.AddCommand(legacyRotateRoot)
+
+	legacyRotateTargets := &cobra.Command{
+		Use:   "rotate-targets <offline-creds.tgz>",
+		Short: "Rotate the offline target signing key for the Factory",
+		Long: `Rotate the offline target signing key for the Factory.
+
+If there are any production targets in your factory - they are re-signed using the new key.
+This command is not allowed if there is an active wave in your factory.`,
+		Deprecated: `it has moved to a new place.
+Instead, please, use the "fioctl keys tuf rotate-offline-key --role=targets --creds=<tuf-creds.tar.gz>".
+`,
+		Hidden:      true,
+		Run:         doRotateOfflineKey,
+		Annotations: map[string]string{tufCmdAnnotation: tufCmdRotateTargetsLegacy},
+		Args:        cobra.ExactArgs(1),
+	}
+	legacyRotateTargets.Flags().StringP("key-type", "k", tufKeyTypeNameRSA, "Key type, supported: Ed25519, RSA (default).")
+	legacyRotateTargets.Flags().StringP("changelog", "m", "", "Reason for doing rotation. Saved in root metadata for tracking change history.")
+	cmd.AddCommand(legacyRotateTargets)
+}
+
+func doRotateOfflineKey(cmd *cobra.Command, args []string) {
+	var (
+		roleName, credsFile string
+		firstTime           bool
+		creds, newCreds     OfflineCreds
+		newKey              *TufSigner
+		signers             []TufSigner
+		err                 error
+	)
+	factory := viper.GetString("factory")
+	keyTypeStr, _ := cmd.Flags().GetString("key-type")
+	keyType := ParseTufKeyType(keyTypeStr)
+	changeLog, _ := cmd.Flags().GetString("changelog")
+	switch cmd.Annotations[tufCmdAnnotation] {
+	case tufCmdRotateOfflineKey:
+		roleName, _ = cmd.Flags().GetString("role")
+		roleName = ParseTufRoleNameOffline(roleName)
+		credsFile, _ = cmd.Flags().GetString("creds")
+		firstTime, _ = cmd.Flags().GetBool("first-time")
+		if firstTime && roleName != tufRoleNameRoot {
+			subcommands.DieNotNil(errors.New("First time TUF key rotation must be performed for the root role."))
+		}
+	case tufCmdRotateRootLegacy:
+		roleName = tufRoleNameRoot
+		credsFile = args[0]
+		firstTime, _ = cmd.Flags().GetBool("initial")
+	case tufCmdRotateTargetsLegacy:
+		roleName = tufRoleNameTargets
+		credsFile = args[0]
+		firstTime = false
```

Comment:
seems like a `default: panic` might not be a bad idea here.

Reply:
https://github.com/foundriesio/fioctl/pull/184/commits/f9cf0788a7ebebc98982ec462e35d3d500931b00 (squash later)